src/metabase/models/query/permissions.clj

```diff
@@ -124,7 +124,7 @@
                                    query)}
                        e)]
         (if throw-exceptions? (throw e) (log/error e)))
-      {:perms/data-access {0 :unrestricted}}))) ; table 0 will never exist
+      {:perms/view-data {0 :unrestricted}}))) ; table 0 will never exist
```

Comment:
we'll probably change `view-data` to be DB-level once we can drop `legacy-no-self-service`. so it might be better to use `:create-queries {0 :query-builder}` here since that'll stay table-level.

the main important thing is that this it's an impossible permission